torch/optim/_multi_tensor/adam.py

```diff
@@ -79,6 +79,8 @@ def step(self, closure=None):
                 if p.grad is not None:
                     if p.grad.is_sparse:
                         raise RuntimeError('Adam does not support sparse gradients, please consider SparseAdam instead')
+                    if group['maximize']:
+                        p.grad = -p.grad
```

Comment:
@ngimel do we have a foreach neg that we would want to use here?

Reply:
Yes, nothing is differentiable here.

Reply:
@OliverFM could you use the foreach version to flip all of them at once?

Reply:
Not sure how this would work best.
Since `_foreach_neg` takes a list, or tuple, of tensor inputs, I'd need to generate this tuple, then apply the function, then loop through the parameters to add the gradients back.

Unless I am missing some smart way of doing a batch assignment of gradients to tensors.

Reply:
You should *not* modify the param's .grad field in this function. So there is no need to add the gradients back.
You can simply generate the gradient list as before and apply this function.

Reply:
made the changes!